### PR TITLE
Cluster

### PR DESCRIPTION
**Pull request by fchirica**
The cluster has been running successfully for 1 week with this code. Tested with the local simulation after performing the merge.

I should better document how to run a cluster: on the master machine, everything but timelord_launcher should run. On the VDF machines, only timelord_launcher should run. If not running a cluster, the scripts should work as before.

### CODE REVIEW

File: lib/chiavdf/fast_vdf/vdf_client.cpp [hunk 1]
```diff
@@ -6,8 +6,10 @@ const int max_length = 2048;
 const int kMaxProcessesAllowed = 3;
 std::mutex socket_mutex;
 
+int process_number;
+
 void PrintInfo(std::string input) {
-    print("VDF Server: " + input);
+    print("VDF Client " + to_string(process_number) + ": " + input);
 }
```
Comment (mariano54) [inline lib/chiavdf/fast_vdf/vdf_client.cpp]: I have changed this file significantly in the integration branch (to reduce the boost dependency), I guess we will deal with that later.

File: scripts/run_all_simulation.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ echo "Note that this simulation will not work if connected to external nodes."
 
 _run_bg_cmd python -m src.server.start_harvester
 _run_bg_cmd python -m src.server.start_timelord
+_run_bg_cmd python -m src.timelord_launcher
```
Comment (mariano54) [inline scripts/run_all_simulation.sh]: Have you tested these scripts?

**Review by mariano54 — APPROVED** [review]
Sorry for the late review. I find the timelord launcher is kind of confusing, since it launches processes that connect to the timelord directly, instead of through the launcher. In the future it might even make more sense to have a long running VDF server in cpp and avoid the launcher altogether (and even combine all VDF servers into the same process). Also I'm kind of concerned with adding too many unnecessary servers, as it increases complexity and things that can go wrong. 

That said, if this is running with no issues in the testing, we can merge it into 1.5.

**Comment by fchirica** [comment]
The reason we go for short lived processes is that it's possible VDF client to have some small memory leftovers, which in time might add up significantly. Killing and spawning the processes frequently protects us from this. Launcher's only job is to keep the CPP alive on the host machine (we could look in the future at other ways to do this, like cron).

File: scripts/run_all_simulation.sh (hunk 2 above)
Comment (fchirica) [reply scripts/run_all_simulation.sh]: I've tested only run_all and run_all_simulation, however others should work exactly the same.

File: lib/chiavdf/fast_vdf/vdf_client.cpp (hunk 1 above)
Comment (fchirica) [reply lib/chiavdf/fast_vdf/vdf_client.cpp]: I'll try to combine your changes in integration and this PR in my new branch, new_vdf, where I implement the new VDF algorithm.

**Pull request merged**